Sources/core/objects/Human.cpp

```diff
@@ -76,12 +78,34 @@ void Human::GenerateInterfaceForFrame()
 
 bool Human::TryMove(Dir direct)
 {
+    VDir pos;
+    if (pulled_object_)
+    {
+        if (!CanTouch(pulled_object_))
+        {
+            pulled_object_ = 0;
+            interface_.UpdatePulling(false);
+        }
+        else
+        {
+            pos.x = GetX() - pulled_object_->GetX();
+            pos.y = GetY() - pulled_object_->GetY();
+        }
+    }
     if (IMob::TryMove(direct))
     {   
         if (owner->GetItem<Shard>().IsValid())
         {
             PlaySoundIfVisible("glass_step.ogg", GetOwner().Id());
         }
+        if (pulled_object_ && !(std::abs(pos.x) + std::abs(pos.y) >= 2))
```

Comment:
It is kind of minor issue, but from my point of view it is better to avoid unnecessary negations where it is possible, because they causes difficulties with understanding of the code.
```
if (pulled_object_ && (std::abs(pos.x) + std::abs(pos.y) < 2))
```
will be a little bit more easy to understand.